tree-sitter-python/test/Main.hs

```diff
@@ -15,13 +15,15 @@ import           TreeSitter.Python
 import qualified TreeSitter.Python.AST as Py
 import           TreeSitter.Unmarshal
 
+-- TODO: add tests that verify correctness for product, sum and leaf types
+
 shouldParseInto :: (MonadIO m, MonadTest m, Unmarshal t, Eq t, Show t) => ByteString -> t -> m ()
 s `shouldParseInto` t = do
   parsed <- liftIO $ parseByteString tree_sitter_python s
   parsed === Right t
 
-pass = Py.PassStatementSimpleStatement (Py.PassStatement {Py.bytes = "pass"})
-one = Py.ExpressionStatementSimpleStatement (Py.ExpressionStatement { Py.bytes = "1" })
+pass = Py.PassStatementSimpleStatement (Py.PassStatement "pass")
+one = Py.ExpressionStatementSimpleStatement (Py.ExpressionStatement [Left (Py.Identifier "1")])
```

Comment:
Does getting this test failing in a new way count as progress? 😬

As @robrix pointed out, `Py.Expression` is a type, not a data constructor but the error message makes it a bit confusing to see when it suggests `Py.Identifier` is valid but incorrect relative to the expected `Py.Expression`:

```
 Couldn't match type ‘Py.Identifier’ with ‘Py.Expression’
      Expected type: Either
                       Py.Expression
                       (Either Py.Assignment (Either Py.AugmentedAssignment Py.Yield))
        Actual type: Either
                       Py.Identifier
                       (Either Py.Assignment (Either Py.AugmentedAssignment Py.Yield))
```

Reply:
Failing in a new way is definitely progress: it gives you parallax on what you’re actually trying to resolve here!

Reply:
Just going to add a note to express what the source of confusion of and what's correct.

I was confused because the error message expresses types that don't mirror the code directly. I tried different permutations of exactly what I have written up there, but I forgot that we concatenate the type onto the constructor, leading to compilation failures and accompanying sadness.